test/opml.test.js

```diff
@@ -0,0 +1,29 @@
+const xmlParser = require('fast-xml-parser');
+const request = require('supertest');
+const app = require('../src/backend/web/app');
+
+describe('GET "/feed/opml" endpoint', () => {
+  let response;
+
+  beforeAll(async () => {
+    response = await request(app).get('/feed/opml');
+  });
+
+  it('should respond with status 200 and content-type text/x-opml', () => {
+    expect(response.status).toEqual(200);
+    expect(response.get('Content-type')).toContain('text/x-opml');
+  });
+
+  it('should respond with a valid XML string that contains all expected elements', () => {
+    expect(xmlParser.validate(response.text)).toBe(true);
+
+    const opmlObj = xmlParser.parse(response.text);
+
+    expect(Object.keys(opmlObj)).toEqual(['opml']);
+    expect(Object.keys(opmlObj.opml)).toEqual(['head', 'body']);
+    expect(Object.keys(opmlObj.opml.head).sort()).toEqual(
+      ['title', 'dateCreated', 'ownerName'].sort()
```

Comment:
am I under the right impression that it all the feeds have a title, dateCreated and ownerName?

Is that from all the different feeds that we post in?

Reply:
This second test ensures that our server responds to GET request sent to our `/feed/opml` endpoint with an OPML structure that looks like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<opml version="2.0">
    <head>
        <title>...</title>
        <dateCreated>...</dateCreated>
        <ownerName>...</ownerName>
    </head>
    <body>...</body>
</opml>
```

The structure of the of each `<outline>` element contained within the `<body>` element, mainly because [`response.text`](https://github.com/Seneca-CDOT/telescope/pull/644/files#diff-2af95cf6b421f595f051f65d42c5383dR18) seems to omit them for some reason, leaving the `<body>` element empty. I would ideally like to test the `<outline>` elements, but I can't seem to GET them... 🤔 

You can get the real OPML output yourself (including all of the `<outline>` elements) by opening http://dev.telescope.cdot.systems/feed/opml in a browser.

Reply:
We probably don't need to test more than this.